src/features/logging/aggregate/index.js

```diff
@@ -15,49 +20,87 @@ export class Aggregate extends AggregateBase {
     this.bufferedLogs = []
     /** held logs during sending, for retries */
     this.outgoingLogs = []
+    /** the estimated bytes of log data waiting to be sent -- triggers a harvest if adding a new log will exceed limit  */
+    this.estimatedBytes = 0
 
     this.#agentRuntime = getRuntime(this.agentIdentifier)
     this.#agentInfo = getInfo(this.agentIdentifier)
 
+    this.harvestTimeSeconds = getConfigurationValue(this.agentIdentifier, 'logging.harvestTimeSeconds')
+
     this.waitForFlags([]).then(() => {
+      this.scheduler = new HarvestScheduler('browser/logs', {
+        onFinished: this.onHarvestFinished.bind(this),
+        retryDelay: this.harvestTimeSeconds,
+        getPayload: this.prepareHarvest.bind(this),
+        raw: true
+      }, this)
+      /** harvest immediately once started to purge pre-load logs collected */
+      this.scheduler.startTimer(this.harvestTimeSeconds, 0)
       /** emitted by instrument class (wrapped loggers) or the api methods directly */
       registerHandler(LOGGING_EVENT_EMITTER_CHANNEL, this.handleLog.bind(this), this.featureName, this.ee)
       this.drain()
     })
   }
 
   handleLog (timestamp, message, attributes, level) {
-    this.bufferedLogs.push(
-      new Log(
-        this.#agentRuntime.timeKeeper.convertRelativeTimestamp(timestamp),
-        message,
-        attributes,
-        level
-      )
+    if (this.blocked) return
+    const log = new Log(
+      this.#agentRuntime.timeKeeper.convertRelativeTimestamp(timestamp),
+      message,
+      attributes,
+      level
     )
+    const logBytes = log.message.length + stringify(log.attributes).length + log.logType.length + 10 // timestamp == 10 chars
```

Comment:
can't assume "message" is actually a string because when customer use `myWrapFn = newrelic.wrapLogger(...)`, the myWrapFn could be passed any type arguments.

Reply:
either stringify if not string type, or leave some paper trail if the "message" turns out not to be a string.

Reply:
Or `.length` would throw an error example

Reply:
stringified